src/rest/handlers/rest_edges.erl

```diff
@@ -0,0 +1,119 @@
+%%%-------------------------------------------------------------------
+%% @doc
+%%  Implements API for manipulating edges.
+%% @end
+%%%-------------------------------------------------------------------
+
+-module(rest_edges).
+-behavior(cowboy_handler).
+
+%% API
+-export([
+    init/2,
+    allowed_methods/2,
+    content_types_accepted/2,
+    is_conflict/2,
+    resource_exists/2
+]).
+
+-export([
+    from_json/2
+]).
+
+
+%%%---------------------------
+%% cowboy_rest callbacks
+%%%---------------------------
+
+init(Req0, State) ->
+    {ParsedParams, Req} =
+        case maps:get(operation, State) of
+            Op when Op =:= add; Op =:= permissions ->
+                {cowboy_req:match_qs([
+                    {from, nonempty}, {to, nonempty}, {permissions, nonempty},
+                    {trace, [], undefined}, {successive, nonempty}
+                ], Req0), Req0};
+            delete ->
+                {cowboy_req:match_qs([
+                    {from, nonempty}, {to, nonempty}, {trace, [], undefined}, {successive, nonempty}
+                ], Req0), Req0};
+            bulk ->
+                {ok, Data, Req1} = cowboy_req:read_body(Req0),
+                ParsedJson = gmm_utils:decode(Data),
+                {ok, Map} = parse_bulk_request(ParsedJson),
+                {#{bulk_request => Map}, Req1}
+        end,
+    NewState = maps:merge(State, ParsedParams),
+    {cowboy_rest, Req, NewState}.
+
+allowed_methods(Req, State) ->
+    {[<<"POST">>], Req, State}.
+
+content_types_accepted(Req, State) ->
+    {[{<<"application/json">>, from_json}], Req, State}.
+
+resource_exists(Req, State) ->
+    #{from := From, to := To} = State,
+    {ok, Result} = graph:edge_exists(From, To),
+    {Result, Req, State}.
+
+is_conflict(Req, State) ->
+    Result = case maps:get(operation, State) of
+                 add -> true;
+                 _ -> false
+             end,
+    {Result, Req, State}.
+
+%% POST handler
+from_json(Req, State) ->
+    Result = case maps:get(operation, State) of
+                 delete ->
+                     #{from := From, to := To} = State,
+                     case graph:edge_exists(From, To) of
+                         {ok, true} -> graph:remove_edge(From, To);
+                         _ -> {error, ""}
+                     end;
+                 permissions ->
+                     #{from := From, to := To, permissions := Permissions} = State,
+                     graph:update_edge(From, To, Permissions);
+                 add ->
+                     #{from := From, to := To, permissions := Permissions} = State,
+                     graph:create_edge(From, To, Permissions);
+                 bulk ->
+                     %% @todo execute bulk request parsed in init/2
+                     ok
+             end,
```

Comment:
Teraz jesteś trochę mało konsekwentny z tymi case'ami, skoro ten który ostatnio zaznaczyłem poszedł do kolejnej linijki. That said, właśnie do takich akcji będzie formater, więc nie będę Cię za to mocno ścigał, tylko zwracam uwagę

Reply:
Jasne, rozumiem o co chodzi, ale zrobiłem to po prostu tak, że do pewnej długości wcięcia zostawiłem tak jak tu, i tyle